src/backends/github/API.js

```diff
@@ -273,19 +273,20 @@ export default class API {
 
   deleteFile(path, message, options={}) {
     const branch = options.branch || this.branch;
+    const pathArray = path.split('/');
+    const filename = last(pathArray);
+    const directory = initial(pathArray).join('/');
+    const fileDataPath = encodeURIComponent(directory);
+    const fileDataURL = `${this.repoURL}/git/trees/${branch}:${fileDataPath}`;
     const fileURL = `${ this.repoURL }/contents/${ path }`;
-    // We need to request the file first to get the SHA
-    return this.request(fileURL, {
-      params: { ref: branch },
-      cache: "no-store",
-    }).then(({ sha }) => this.request(fileURL, {
-      method: "DELETE",
-      params: {
-        sha,
-        message,
-        branch,
-      },
-    }));
+
+    // We need to request the tree first to get the SHA
+    return this.request(fileDataURL, { cache: 'no-store' })
```

Comment:
We were downloading the file to get it's SHA, but the `contents` endpoint has a 1MB limit for GET requests, so we use the `trees` endpoint to get the blob data direct instead.